plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -156,7 +150,10 @@ public DefaultGlueColumnStatisticsProvider(AWSGlueAsync glueClient, String catal
             }
             return columnStatsMapBuilder.build();
         }
-        catch (AmazonServiceException ex) {
+        catch (EntityNotFoundException ex) {
```

Comment:
@losipiuk should we fail if Trino tries to fetch statistics for a partition which doesn't exist? Or return empty stats?

Reply:
Btw, this was changed to avoid catching the `TrinoException(HIVE_METASTORE_ERROR, EntityNotFoundException)` in the caller to detect that a partition doesn't exist.

I'd prefer if this responsibility of handling `EntityNotFoundException` would be given to the caller since this is a public method and part of the API.

WDYT @losipiuk ?